### ISSUE TITLE
ClosedIllegalStateException in RemoteConnector if socketReconnector is triggered

### ISSUE THREAD

**Issue by alamar** [OpenHFT/Chronicle-Network#155]
HostConnector installs the following `socketReconnector` on `NetworkContext`:
```
                .socketReconnector(new ClosableRunnable() {
                    @Override
                    public void run() {
                        close();
                        if (!eventLoop.isClosing())
                            connect();
                    }

    public synchronized void close() {
        ...
        closeQuietly(nc);
        this.nc = null;
    }
```

If it is triggered while `RCEventHandler` is opening connection, it is possible to get the following unhandled exception:
```
2021-11-18T00:13:45.140 [main/map-cluster-0-blocking-event-loop-11] WARN  n.o.c.n.RemoteConnector$RCEventHandler - 452dc273 threw
net.openhft.chronicle.core.io.ClosedIllegalStateException: software.chronicle.enterprise.map.cluster.MapClusterNetworkContext closed
        at net.openhft.chronicle.core.io.AbstractCloseable.throwExceptionIfClosedInSetter(AbstractCloseable.java:259)
        at net.openhft.chronicle.network.VanillaNetworkContext.throwExceptionIfClosedInSetter(VanillaNetworkContext.java:80)
        at net.openhft.chronicle.network.VanillaNetworkContext.socketChannel(VanillaNetworkContext.java:72)
        at net.openhft.chronicle.network.RemoteConnector$RCEventHandler.action(RemoteConnector.java:172)
        at net.openhft.chronicle.threads.BlockingEventLoop$Runner.run(BlockingEventLoop.java:180)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at net.openhft.chronicle.core.threads.CleaningThread.run(CleaningThread.java:104)
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Network-2.22ea6](https://github.com/OpenHFT/Chronicle-Network/releases/chronicle-network-2.22ea6), [BOM-2.22ea61](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea61)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Network-2.21.94](https://github.com/OpenHFT/Chronicle-Network/releases/chronicle-network-2.21.94), [BOM-2.21.243](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.21.243)
